app/code/Magento/Catalog/Model/Product/Url.php

```diff
@@ -162,11 +162,8 @@ public function getUrl(\Magento\Catalog\Model\Product $product, $params = [])
                     \Magento\Store\Model\ScopeInterface::SCOPE_STORE
                 );
 
-                if ($categoryId) {
-                    $filterData[UrlRewrite::METADATA]['category_id'] = $categoryId;
-                } elseif (!$useCategories) {
-                    $filterData[UrlRewrite::METADATA]['category_id'] = '';
-                }
+                $filterData[UrlRewrite::METADATA]['category_id']
```

Comment:
Hi @atishgoswami. Thank you for your contribution.
That's a very good point. If the corresponding configuration is not enabled, the system should not take into account a category ID